var/spack/repos/builtin/packages/esmf/package.py

```diff
@@ -75,7 +63,9 @@ class Esmf(MakefilePackage):
     depends_on("netcdf-fortran@3.6:", when="+netcdf")
     depends_on("parallel-netcdf@1.2.0:", when="+pnetcdf")
     depends_on("xerces-c@3.1.0:", when="+xerces")
-    depends_on("parallelio@2.5.8:", when="+parallelio")
+    depends_on("parallelio@2.5.7:", when="@8.3.0:8.3.99+external-parallelio")
```

Comment:
It is good to have those. Thanks. It was fixed to 2.5.8 before for all the versions.